contracts/modules/TransferManager/BlacklistTransferManager.sol

```diff
@@ -0,0 +1,384 @@
+pragma solidity ^0.4.24;
+
+import "./ITransferManager.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title Transfer Manager module to automate blacklist and restrict transfers
+ */
+contract BlacklistTransferManager is ITransferManager {
+    using SafeMath for uint256;
+
+    bytes32 public constant ADMIN = "ADMIN";
+    
+    struct BlacklistsDetails {
+        uint256 startTime;
+        uint256 endTime;
+        uint256 repeatPeriodTime;
+    }
+
+    //hold the different blacklist details corresponds to its name
+    mapping(bytes32 => BlacklistsDetails) public blacklists;
+
+    //hold the different name of blacklist corresponds to a investor
+    mapping(address => bytes32[]) investorToBlacklist;
+
+    //get list of the addresses for a particular blacklist
+    mapping(bytes32 => address[]) blacklistToInvestor;
+
+    //mapping use to store the indexes for different blacklist types for a investor
+    mapping(address => mapping(bytes32 => uint256)) investorToIndex;
+
+    //mapping use to store the indexes for different investor for a blacklist type
+    mapping(bytes32 => mapping(address => uint256)) blacklistToIndex;
+
+    bytes32[] allBlacklists;
+   
+    // Emit when new blacklist type is added
+    event AddBlacklistType(
+        uint256 _startTime, 
+        uint256 _endTime, 
+        bytes32 _name, 
+        uint256 _repeatPeriodTime
+    );
+    
+    // Emit when there is a change in the blacklist type
+    event ModifyBlacklistType(
+        uint256 _startTime,
+        uint256 _endTime, 
+        bytes32 _name, 
+        uint256 _repeatPeriodTime
+    );
+    
+    // Emit when the added blacklist type is deleted
+    event DeleteBlacklistType(
+        bytes32 _name
+    );
+
+    // Emit when new investor is added to the blacklist type
+    event AddInvestorToBlacklist(
+        address indexed _investor, 
+        bytes32 _blacklistName
+    );
+    
+    // Emit when investor is deleted from the blacklist type
+    event DeleteInvestorFromBlacklist(
+        address indexed _investor,
+        bytes32 _blacklist
```

Comment:
Blacklist name has a different names on each event (_name, _blacklistName, _blacklist). We should unify them.